feat: notification badge component (#267)
Atom-Learning/components: src/components/notification-badge/NotificationBadge.mdx

---
title: NotificationBadge
component: NotificationBadge
description: A badge that wraps content and displays a notification
category: Layout
---

The NotificationBadge component is a lightweight wrapper for content such as ActionIcons, but can be wrapped around anything.

It displays a badge in top right of the content it is wrapping, which displays a `value` passed in as a prop. This could be, for example, the number of filters selected on a select filters badge.

```tsx preview
<NotificationBadge value={3}>
  <ActionIcon appearance="outline" size="lg" isRounded>
    <Icon is={Controls} />
  </ActionIcon>
</NotificationBadge>
```

## Value

Whilst you would normally want to display a number, the `value` prop also supports strings.

> ## Themes

Author: really good point - will change now :)

---

> title: NotificationBadge

Reviewer: ```suggestion
title: Notification Badge
```

Author: oops - changed :)

component: NotificationBadge
description: A badge that wraps content and displays a notification

> category: Layout

Reviewer: ```suggestion
category: Feedback
```

I think this is a more appropriate category

Author: yeah definitely, good catch

---

> The NotificationBadge component is a lightweight wrapper for content such as ActionIcons, but can be wrapped around anything.

Reviewer: ```suggestion
The `NotificationBadge` component is a lightweight wrapper for content such as an `ActionIcon` but can be wrapped around any block-level element.
```

Author: again nice catch, good to be specific :)

It displays a badge in top right of the content it is wrapping, which displays a `value` passed in as a prop. This could be, for example, the number of filters selected on a select filters badge.

```tsx preview
<NotificationBadge value={3}>
  <ActionIcon appearance="outline" size="lg" isRounded>
    <Icon is={Controls} />
  </ActionIcon>
</NotificationBadge>
```

## Value

Whilst you would normally want to display a number, the `value` prop also supports strings.

## Themes

The component accepts a `theme` prop of either `primary`, `dark`. Default is `primary`.

```tsx preview
<NotificationBadge value={1}>
  <ActionIcon appearance="outline" size="lg" isRounded>
    <Icon is={Controls} />
  </ActionIcon>
</NotificationBadge>
<NotificationBadge theme="dark" value={2}>
  <ActionIcon appearance="outline" size="lg" isRounded>
    <Icon is={Controls} />

>   </ActionIcon>

Reviewer: I would use the documentation as a good opportunity to show different combinations, maybe we could render a `Button` here?

Reviewer: Oh and different values as well, e.g. `value={64}`

Author: good idea :)
